src/tests/AttributesManagersTest.cpp

```diff
@@ -163,6 +165,43 @@ class AttributesManagersTest : public testing::Test {
 
   }  // AttributesManagersTest::testCreateAndRemove
 
+  /**
+   * @brief Test creation, copying and removal of templates for lights
+   * attributes managers.
+   * @param mgr the Attributes Manager being tested,
+   * @param handle the handle of the desired attributes template to work with
+   */
+
+  void testCreateAndRemoveLights(
+      AttrMgrs::LightLayoutAttributesManager::ptr mgr,
+      const std::string& handle) {
+    // meaningless key to modify attributes for verifcation of behavior
+    std::string keyStr = "tempKey";
+    // get starting number of templates
+    int orignNumTemplates = mgr->getNumObjects();
+
+    // Source config for lights holds multiple light configurations.
+    // Create a single template for each defined light in configuration and
+    // register it.
+    mgr->createObject(handle, true);
+    // get number of templates loaded
+    int numLoadedLights = mgr->getNumObjects();
+    // verify lights were added
+    ASSERT_NE(numLoadedLights, orignNumTemplates);
+
+    // get handles of all lights added
+    auto lightHandles = mgr->getObjectHandlesBySubstring();
```

Comment:
This call to `getObjectHandlesBySubstring` with no arguments means "get all handles", yes? That's a little confusing.

Reply:
yeah, defaults to all handles (every string has empty string :) )

Reply:
All of the attributes managers' functionality is exposed in python via bindings that have doc/popup to help with this.